### PR TITLE
Generate title and description with Llava13b

### PR DESCRIPTION
**Pull request by microchipgnu**
This PR adds [Replicate](https://www.replicate.com) support to Minsta.

I'm following this standalone implementation I did a few days ago and adding it into this project. See more https://github.com/microchipgnu/next-replicate

I've also refactored the `_mintImage` method in App Provider into a separated hook. This way we clean up the provider (it was getting bloated) and have more readable and composable hook.

🚨 **Before merging** we need to make sure the photo in base64 is <10MB, maybe we can decrease the resolution first before sending it to Replicate. EDIT: Added a method that decreases the resolution

### CODE REVIEW

**Review by rubenmarcus — APPROVED** [review]

File: src/utils/useMint.tsx [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+              args: {
+                metadata: JSON.stringify(metadata),
+                nft_contract_id: constants.tokenContractAddress,
+              },
+              gas: "200000000000000",
+              deposit: "10000000000000000000000",
+            },
+          },
+        ],
+        callbackUrl: baseUrl,
```
Comment (rubenmarcus) [inline src/utils/useMint.tsx:90]: this wont be needed as pass the callbackUrl globally on the provider on #52 


check 
https://github.com/Mintbase/minsta/blob/db09c758503bbb9572db861da3ae1904838907b4/src/providers/providers.tsx

File: src/utils/useMint.tsx [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+      console.error("Failed to upload reference:", error);
+      setLoading(false);
+      throw new Error("Failed to upload reference");
+    }
+  };
+
+  const performTransaction = async (
+    wallet: any,
+    metadata: any,
+    baseUrl: string
```
Comment (rubenmarcus) [inline src/utils/useMint.tsx:66]: no need for baseUrl

File: src/utils/useMint.tsx [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+      const photoFile = convertBase64ToFile(photo);
+      const replicatePhoto = await reduceImageSize(photo, 10); //10MB limit replicate
+      const titleAndDescription = await getTitleAndDescription(replicatePhoto);
+      const refObject = {
+        title: titleAndDescription.title,
+        description: titleAndDescription.description,
+        media: photoFile,
+      };
+      const uploadedData = await uploadReferenceObject(refObject);
+      const baseUrl = `${window.location.origin}/`;
```
Comment (rubenmarcus) [inline src/utils/useMint.tsx:177]: no need. for baseUrl

**Review by rubenmarcus — CHANGES_REQUESTED** [review]

File: src/utils/useMint.tsx (hunk 1 above)
Comment (microchipgnu) [reply src/utils/useMint.tsx:90]: https://github.com/Mintbase/minsta/blob/4d978c82edea2554e7c2a366026f15f8d492bb49/src/providers/app.tsx#L125

This is in prod right now too. I've refactored based on this file alone, shall I update this accordingly?

**Review by rubenmarcus — APPROVED** [review]

**Pull request merged**